temporal-sdk/src/main/java/io/temporal/worker/tuning/ResourceBasedSlotSupplier.java

```diff
@@ -32,6 +33,18 @@ public class ResourceBasedSlotSupplier<SI extends SlotInfo> implements SlotSuppl
   private final ResourceBasedController resourceController;
   private final ResourceBasedSlotOptions options;
   private Instant lastSlotIssuedAt = Instant.EPOCH;
+  // For slot reservations that are waiting to re-check resource usage
+  private static final ScheduledExecutorService scheduler =
```

Comment:
I think you should consider accepting this in the constructor. Can have a default, but we should allow people to control thread creation if they want.